sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveUDFSuite.scala

```diff
@@ -487,6 +489,28 @@ class HiveUDFSuite extends QueryTest with TestHiveSingleton with SQLTestUtils {
     assert(count4 == 1)
     sql("DROP TABLE parquet_tmp")
   }
+
+  test("Hive Stateful UDF") {
+    withUserDefinedFunction("statefulUDF" -> true, "statelessUDF" -> true) {
+      sql(s"CREATE TEMPORARY FUNCTION statefulUDF AS '${classOf[StatefulUDF].getName}'")
+      sql(s"CREATE TEMPORARY FUNCTION statelessUDF AS '${classOf[StatelessUDF].getName}'")
+      val testData = spark.range(10).repartition(1)
+      val testData1 = spark.range(10).repartition(2)
+
+      // Expected Max(s) is 10 as statefulUDF returns the sequence number starting from 1.
+      checkAnswer(testData.selectExpr("statefulUDF() as s").agg(max($"s")), Row(10))
+
+      // Expected Max(s) is 10 as statefulUDF returns the sequence number starting from 1.
+      // Note that testData.union(testData) will have Row(20) because internally it is converted
```

Comment:
if you need to write comments to explain some technical problems for the test, it would be better to rewrite the test and avoid this problem. How about:
```
checkAnswer(spark.range(10).repartition(2, $"id" < 5).selectExpr("statefulUDF() as s").agg(max($"s")), Row(5))
```